httpcore/_backends/base.py

```diff
@@ -39,6 +39,28 @@ def lookup_sync_backend(name: str) -> "SyncBackend":
     return SyncBackend()
 
 
+class AsyncDatagramSocket:
+    """
+    A datagram socket with read/write operations. Abstracts away any asyncio-specific
+    interfaces into a more generic base class, that we can use with alternate
+    backends, or for stand-alone test cases.
+    """
+
+    remote_addr: SocketAddr
+
+    def get_http_version(self) -> str:
+        return "HTTP/3"
+
+    async def aclose(self) -> None:
+        raise NotImplementedError()  # pragma: no cover
+
+    async def receive(self) -> Tuple[bytes, SocketAddr]:
```

Comment:
For consistency, should these be named read() and write()? Otherwise we might consider to update the doc string for this class to use the same receive/send lingo.

Reply:
I don't think read / write make sense here as we are also conveying another information: the remote address. The "classical" corresponding APIs are `recvfrom` and `sendto` but as I understand we're avoid abbreviations, I went for "receive" and "send". 

`anyio` seems to have settled on the same names: https://github.com/agronholm/anyio/blob/3a0b805a3f4410009f2b6610d329fb9cc2b90e5a/src/anyio/_backends/_asyncio.py#L1258